packages/react-router-dom/index.tsx

```diff
@@ -632,12 +636,14 @@ export function RouterProvider({
 
 function DataRoutes({
   routes,
+  future,
   state,
 }: {
   routes: DataRouteObject[];
+  future: RemixRouter["future"];
   state: RouterState;
 }): React.ReactElement | null {
-  return useRoutesImpl(routes, undefined, state);
+  return useRoutesImpl(routes, undefined, state, future);
```

Comment:
Plumb down the `router.future` from RouterProvider so we can use it in `renderMatches` and only do the fallback stuff when enabled